### ISSUE TITLE
ZSTD_c_experiencelParam6 issue not found

### ISSUE THREAD

**Issue by wk1583241477** [gyscos/zstd-rs#270]
![image](https://github.com/gyscos/zstd-rs/assets/46584343/8d8ffe8e-6157-49d8-af79-d5cc142b1ef3)

The error message is as follows：

error[E0432]: unresolved import `zstd_sys::ZSTD_cParameter::ZSTD_c_experimentalParam6`
   --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zstd-safe-6.0.6/src/lib.rs:609:13
    |
609 |             ZSTD_c_experimentalParam6 as ZSTD_c_targetCBlockSize,
    |             -------------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             no `ZSTD_c_experimentalParam6` in `ZSTD_cParameter`
    |             help: a similar name exists in the module: `ZSTD_c_experimentalParam1`
 
make[3]: Leaving directory '/home/wk/milvus/cmake_build'afe                                                                                                  
[ 60%] Built target knowhere
make[3]: Entering directory '/home/wk/milvus/cmake_build'
For more information about this error, try `rustc --explain E0432`.
error: could not compile `zstd-safe` (lib) due to previous error

**Comment by nickkaltner** [comment]
for anyone encountering this change, you can do the following to work around it;

```
[dependencies]
zstd-sys = "=2.0.9"
```

which locks the compatible version

**Comment by DragisBlade** [comment]
It is not only that the Build is broken, hence this build is official and the newest available version. So auomatic updates will download it!

Resulting in a major issue for all users doing
`cargo update`

For me it drawed the broken build as a deep dependency of tantivy today and it took me quite a while to find this issue-post!

**Comment by frnsys** [comment]
Really frustrating that such a breaking change was introduced.

For others who encounter this problem because of a transitive dependency, I added the following to my `Cargo.toml`:

```
[patch.crates-io]
zstd-sys = { git = "https://github.com/gyscos/zstd-rs.git", rev = "1779b385b42b08f958b767a37878dfa6a0b4f6a4" }
```
and then ran `cargo update`. This will use the commit before the change.

**Comment by gyscos** [comment]
To be clear, it only happens when all of these situations combine:
* An old version of zstd-safe is used
* the experimental/unstable flag is explicitly requested - this acts as a deliberate choice to deal with such changes
* The version of zstd-sys was not locked, either through Cargo.lock or a simple exact version pin in de dependencies (no need for a patch.crates-io section).
For example, this would lock the version:
```
[dependencies]
zstd-sys = "=2.0.9"
```

**Comment by BSteffaniak** [comment]
Yes, the cause of this for me seems to be tantivy as well. It looks like this issue is related: https://github.com/quickwit-oss/tantivy/issues/2339

**Comment by BSteffaniak** [comment]
The [issue with Tantivy](https://github.com/quickwit-oss/tantivy/issues/2339) has been fixed in [version 0.22.0](https://github.com/quickwit-oss/tantivy/releases/tag/v0.22.0) (released about 10 hours ago). That version resolves this issue for me.

**Comment by stevemk14ebr** [comment]
The tantivy release has some breaking changes FYI
